MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc

```diff
@@ -0,0 +1,64 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/FWLite/interface/Record.h"
+
+#include "MagneticField/ParametrizedEngine/interface/alpaka/ParabolicParametrizedMagneticField.h"
+
+#include <iostream>
+#include <fstream>
+#include <Eigen/Core>
+
+using namespace edm;
+using namespace std;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE::MagneticFieldParabolicPortable;
+using Vector3f = Eigen::Matrix<float, 3, 1>;
+
+int main() {
+  ifstream file;
+  edm::FileInPath mydata("MagneticField/Engine/data/Regression/referenceField_160812_RII_3_8T.bin");
+  file.open(mydata.fullPath().c_str(), ios::binary);
+
+  float resolution = 0.0001;
+  float maxdelta = 0.;
+  int fail = 0;
+  int count = 0;
+
+  float px, py, pz;
+  float bx, by, bz;
+  GlobalPoint gp;
+
```

Comment:
Here's a suggestion how to reorganise the test to actually run the computation on GPUs:
  - read the first `numberOfPoints` valid points into a vector, array, or SoA
  - copy them to the device
  - submit a kernel that calls `MagneticFieldAtPoint` for all those points and store the result in global memory
  - copy the result back to the host
  - compare the results with the reference values

Reply:
@fwyzard, I implemented most of the changes (https://github.com/cms-sw/cmssw/pull/44360/commits/05933ccae9d54ebc5c7bb8a674204fb6102f4252) but I have a couple of compilation errors at
`>> Compiling  src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc`:

1.
```
src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc:69:71: error: no matching function for call to 'make_host_buffer<float [], template<class T> using Platform = typename alpaka::trait::PlatformType<T, void>::type>(const size_t&)'
   69 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
```
2.
```
src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc:82:64: error: no matching function for call to 'devices<template<class T> using Platform = typename alpaka::trait::PlatformType<T, void>::type>()'
   82 |   for (auto const& device : cms::alpakatools::devices<Platform>()) {
``` 
3.
```
src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticField.cc:93:51: error: 'Acc1D' was not declared in this scope
   93 |     auto workDiv = cms::alpakatools::make_workdiv<Acc1D>(1, size);
      |                                                   ^~~~~ 

```
so I guess some `#include` is missing, but I have all of the `#include`s of `HeterogeneousCore/AlpakaInterface/test/alpaka/testIndependentKernel.dev.cc`

Reply:
See Matti's [comment](https://github.com/cms-sw/cmssw/pull/44360/files#r1670640121) above.

Reply:
Also, it helps if you paste the actual error messages, not only the line that causes them.

Reply:
I moved the file in the `alpaka` subdirectory but the errors did not change:
```
[lecriste@lxplus912 ParametrizedEngine]$ scram b -j
>> Local Products Rules ..... started
>> Local Products Rules ..... done
>> Entering Package MagneticField/ParametrizedEngine
>> Creating project symlinks
>> Compiling alpaka/cuda src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc
>> Compiling alpaka/rocm src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc
>> Compiling alpaka/serial src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc: In function 'int main()':
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:43: error: parse error in template argument list
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                                           ^~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:60: error: no matching function for call to 'devices<<expression error> >()'
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/CachingAllocator.h:18,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/getDeviceCachingAllocator.h:11,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/CachedBufAlloc.h:6,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:9,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:15:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/devices.h:22:53: note: candidate: 'template<class TPlatform, class> const std::vector<typename alpaka::trait::DevType<T>::type>& cms::alpakatools::devices()'
   22 |   inline std::vector<alpaka::Dev<TPlatform>> const& devices() {
      |                                                     ^~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/devices.h:22:53: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:60: error: template argument 1 is invalid
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:36: error: parse error in template argument list
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: no matching function for call to 'make_host_buffer<float [], <expression error> >(const size_t&)'
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note: candidate: 'template<class T> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   65 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: candidate: 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: wrong number of template arguments (2, should be 1)
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: provided for 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note: candidate: 'template<class T> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   77 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   85 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   92 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   99 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note: candidate: 'template<class T, class TQueue> std::enable_if_t<(isQueue<TQueue> && (! is_array_v<T>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  109 |   std::enable_if_t<alpaka::isQueue<TQueue> and not std::is_array_v<T>, host_buffer<T>> make_host_buffer(
      |                                                                                        ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_unbounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&, alpaka_common::Extent)'
  123 |   make_host_buffer(TQueue const& queue, Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note:   candidate expects 2 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_bounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  137 |   make_host_buffer(TQueue const& queue) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:37: error: parse error in template argument list
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: no matching function for call to 'make_host_buffer<float [], <expression error> >(const size_t&)'
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note: candidate: 'template<class T> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   65 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: candidate: 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: wrong number of template arguments (2, should be 1)
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: provided for 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note: candidate: 'template<class T> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   77 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   85 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   92 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   99 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note: candidate: 'template<class T, class TQueue> std::enable_if_t<(isQueue<TQueue> && (! is_array_v<T>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  109 |   std::enable_if_t<alpaka::isQueue<TQueue> and not std::is_array_v<T>, host_buffer<T>> make_host_buffer(
      |                                                                                        ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_unbounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&, alpaka_common::Extent)'
  123 |   make_host_buffer(TQueue const& queue, Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note:   candidate expects 2 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_bounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  137 |   make_host_buffer(TQueue const& queue) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:93:18: error: reference to 'Queue' is ambiguous
   93 |     auto queue = Queue(device);
      |                  ^~~~~
In file included from /afs/cern.ch/work/l/lecriste/Electron_seeding/CMSSW_14_1_0_pre0_alpaka_noCheckdeps/src/MagneticField/ParametrizedEngine/interface/alpaka/ParabolicParametrizedMagneticField.h:8,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:11:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/config.h:138:9: note: candidates are: 'using Queue = using QueueCpuBlocking = class alpaka::QueueGenericThreadsBlocking<alpaka::DevCpu>'
  138 |   using Queue = alpaka::QueueCpuBlocking;
      |         ^~~~~
In file included from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/kernel/Traits.hpp:15,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/acc/Traits.hpp:14,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/acc/AccCpuOmp2Blocks.hpp:27,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/alpaka.hpp:13,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:1:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/queue/Traits.hpp:70:11: note:                 'template<class TEnv, class TProperty> using Queue = typename alpaka::trait::QueueType<TEnv, TProperty>::type'
   70 |     using Queue = typename trait::QueueType<TEnv, TProperty>::type;
      |           ^~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:104:60: error: expected primary-expression before ',' token
  104 |     alpaka::exec<Acc1D>(queue, workDiv, MagneticFieldKernel, points_dev.data(), deviceB.data(), size);
      |                                                            ^
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc: In function 'int main()':
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:43: error: parse error in template argument list
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                                           ^~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:60: error: no matching function for call to 'devices<<expression error> >()'
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/CachingAllocator.h:18,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/getDeviceCachingAllocator.h:11,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/CachedBufAlloc.h:6,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:9,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:15:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/devices.h:22:53: note: candidate: 'template<class TPlatform, class> const std::vector<typename alpaka::trait::DevType<T>::type>& cms::alpakatools::devices()'
   22 |   inline std::vector<alpaka::Dev<TPlatform>> const& devices() {
      |                                                     ^~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/devices.h:22:53: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:60: error: template argument 1 is invalid
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:36: error: parse error in template argument list
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: no matching function for call to 'make_host_buffer<float [], <expression error> >(const size_t&)'
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note: candidate: 'template<class T> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   65 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: candidate: 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: wrong number of template arguments (2, should be 1)
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: provided for 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note: candidate: 'template<class T> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   77 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   85 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   92 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   99 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note: candidate: 'template<class T, class TQueue> std::enable_if_t<(isQueue<TQueue> && (! is_array_v<T>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  109 |   std::enable_if_t<alpaka::isQueue<TQueue> and not std::is_array_v<T>, host_buffer<T>> make_host_buffer(
      |                                                                                        ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_unbounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&, alpaka_common::Extent)'
  123 |   make_host_buffer(TQueue const& queue, Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note:   candidate expects 2 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_bounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  137 |   make_host_buffer(TQueue const& queue) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:37: error: parse error in template argument list
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: no matching function for call to 'make_host_buffer<float [], <expression error> >(const size_t&)'
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note: candidate: 'template<class T> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   65 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: candidate: 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: wrong number of template arguments (2, should be 1)
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: provided for 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note: candidate: 'template<class T> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   77 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   85 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   92 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   99 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note: candidate: 'template<class T, class TQueue> std::enable_if_t<(isQueue<TQueue> && (! is_array_v<T>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  109 |   std::enable_if_t<alpaka::isQueue<TQueue> and not std::is_array_v<T>, host_buffer<T>> make_host_buffer(
      |                                                                                        ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_unbounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&, alpaka_common::Extent)'
  123 |   make_host_buffer(TQueue const& queue, Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note:   candidate expects 2 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_bounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  137 |   make_host_buffer(TQueue const& queue) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:93:18: error: reference to 'Queue' is ambiguous
   93 |     auto queue = Queue(device);
      |                  ^~~~~
In file included from /afs/cern.ch/work/l/lecriste/Electron_seeding/CMSSW_14_1_0_pre0_alpaka_noCheckdeps/src/MagneticField/ParametrizedEngine/interface/alpaka/ParabolicParametrizedMagneticField.h:8,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:11:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/config.h:48:9: note: candidates are: 'using Queue = using QueueCudaRtNonBlocking = alpaka::QueueUniformCudaHipRtNonBlocking<alpaka::ApiCudaRt>'
   48 |   using Queue = alpaka::QueueCudaRtNonBlocking;
      |         ^~~~~
In file included from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/kernel/Traits.hpp:15,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/acc/Traits.hpp:14,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/acc/AccCpuOmp2Blocks.hpp:27,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/alpaka.hpp:13,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:1:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/queue/Traits.hpp:70:11: note:                 'template<class TEnv, class TProperty> using Queue = typename alpaka::trait::QueueType<TEnv, TProperty>::type'
   70 |     using Queue = typename trait::QueueType<TEnv, TProperty>::type;
      |           ^~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:104:60: error: expected primary-expression before ',' token
  104 |     alpaka::exec<Acc1D>(queue, workDiv, MagneticFieldKernel, points_dev.data(), deviceB.data(), size);
      |                                                            ^
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc: In function 'int main()':
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:43: error: parse error in template argument list
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                                           ^~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:60: error: no matching function for call to 'devices<<expression error> >()'
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
In file included from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/CachingAllocator.h:18,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/getDeviceCachingAllocator.h:11,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/CachedBufAlloc.h:6,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:9,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:15:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/devices.h:22:53: note: candidate: 'template<class TPlatform, class> const std::vector<typename alpaka::trait::DevType<T>::type>& cms::alpakatools::devices()'
   22 |   inline std::vector<alpaka::Dev<TPlatform>> const& devices() {
      |                                                     ^~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/devices.h:22:53: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:44:60: error: template argument 1 is invalid
   44 |   auto const& devices = cms::alpakatools::devices<Platform>();
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:36: error: parse error in template argument list
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: no matching function for call to 'make_host_buffer<float [], <expression error> >(const size_t&)'
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note: candidate: 'template<class T> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   65 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: candidate: 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: wrong number of template arguments (2, should be 1)
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: provided for 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note: candidate: 'template<class T> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   77 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   85 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   92 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   99 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note: candidate: 'template<class T, class TQueue> std::enable_if_t<(isQueue<TQueue> && (! is_array_v<T>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  109 |   std::enable_if_t<alpaka::isQueue<TQueue> and not std::is_array_v<T>, host_buffer<T>> make_host_buffer(
      |                                                                                        ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_unbounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&, alpaka_common::Extent)'
  123 |   make_host_buffer(TQueue const& queue, Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note:   candidate expects 2 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_bounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  137 |   make_host_buffer(TQueue const& queue) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:79:71: error: template argument 2 is invalid
   79 |   auto in_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:37: error: parse error in template argument list
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: no matching function for call to 'make_host_buffer<float [], <expression error> >(const size_t&)'
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note: candidate: 'template<class T> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   65 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:65:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: candidate: 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: wrong number of template arguments (2, should be 1)
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:71:3: note: provided for 'template<class T> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   71 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note: candidate: 'template<class T> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   77 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:77:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(! is_array_v<T>), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   85 |   std::enable_if_t<not std::is_array_v<T>, host_buffer<T>> make_host_buffer() {
      |                                                            ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:85:60: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_unbounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(alpaka_common::Extent)'
   92 |   make_host_buffer(Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:92:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note: candidate: 'template<class T, class TPlatform> std::enable_if_t<(is_bounded_array_v<T> && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer()'
   99 |   make_host_buffer() {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:99:3: note:   candidate expects 0 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note: candidate: 'template<class T, class TQueue> std::enable_if_t<(isQueue<TQueue> && (! is_array_v<T>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  109 |   std::enable_if_t<alpaka::isQueue<TQueue> and not std::is_array_v<T>, host_buffer<T>> make_host_buffer(
      |                                                                                        ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:109:88: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_unbounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&, alpaka_common::Extent)'
  123 |   make_host_buffer(TQueue const& queue, Extent extent) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:123:3: note:   candidate expects 2 arguments, 1 provided
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note: candidate: 'template<class T, class TQueue> std::enable_if_t<((isQueue<TQueue> && is_bounded_array_v<T>) && (! is_array_v<typename std::remove_extent<_Tp>::type>)), typename cms::alpakatools::detail::buffer_type<alpaka::DevCpu, T>::type> cms::alpakatools::make_host_buffer(const TQueue&)'
  137 |   make_host_buffer(TQueue const& queue) {
      |   ^~~~~~~~~~~~~~~~
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/memory.h:137:3: note:   template argument deduction/substitution failed:
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:80:72: error: template argument 2 is invalid
   80 |   auto out_host = cms::alpakatools::make_host_buffer<float[], Platform>(size);
      |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:93:18: error: reference to 'Queue' is ambiguous
   93 |     auto queue = Queue(device);
      |                  ^~~~~
In file included from /afs/cern.ch/work/l/lecriste/Electron_seeding/CMSSW_14_1_0_pre0_alpaka_noCheckdeps/src/MagneticField/ParametrizedEngine/interface/alpaka/ParabolicParametrizedMagneticField.h:8,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:11:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/HeterogeneousCore/AlpakaInterface/interface/config.h:93:9: note: candidates are: 'using Queue = using QueueHipRtNonBlocking = alpaka::QueueUniformCudaHipRtNonBlocking<alpaka::ApiHipRt>'
   93 |   using Queue = alpaka::QueueHipRtNonBlocking;
      |         ^~~~~
In file included from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/kernel/Traits.hpp:15,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/acc/Traits.hpp:14,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/acc/AccCpuOmp2Blocks.hpp:27,
                 from /cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/alpaka.hpp:13,
                 from src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:1:
/cvmfs/cms.cern.ch/el9_amd64_gcc12/external/alpaka/1.1.0-d5100a17ef5e9f41a14b52000051b265/include/alpaka/queue/Traits.hpp:70:11: note:                 'template<class TEnv, class TProperty> using Queue = typename alpaka::trait::QueueType<TEnv, TProperty>::type'
   70 |     using Queue = typename trait::QueueType<TEnv, TProperty>::type;
      |           ^~~~~
src/MagneticField/ParametrizedEngine/test/alpaka/testParabolicParametrizedMagneticField.cc:104:60: error: expected primary-expression before ',' token
  104 |     alpaka::exec<Acc1D>(queue, workDiv, MagneticFieldKernel, points_dev.data(), deviceB.data(), size);
      |                                                            ^
gmake: *** [config/SCRAM/GMake/Makefile.rules:1824: tmp/el9_amd64_gcc12/src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticFieldSerialSync/alpaka/testParabolicParametrizedMagneticField.cc.o] Error 1
gmake: *** Waiting for unfinished jobs....
gmake: *** [config/SCRAM/GMake/Makefile.rules:1824: tmp/el9_amd64_gcc12/src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticFieldCudaAsync/alpaka/testParabolicParametrizedMagneticField.cc.o] Error 1
gmake: *** [config/SCRAM/GMake/Makefile.rules:1824: tmp/el9_amd64_gcc12/src/MagneticField/ParametrizedEngine/test/testParabolicParametrizedMagneticFieldROCmAsync/alpaka/testParabolicParametrizedMagneticField.cc.o] Error 1
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```